mypy/semanal.py

```diff
@@ -3230,10 +3237,16 @@ def name_not_defined(self, name: str, ctx: Context) -> None:
                 self.add_fixture_note(fullname, ctx)
 
     def name_already_defined(self, name: str, ctx: Context,
-                             original_ctx: Optional[SymbolTableNode] = None) -> None:
+                             original_ctx: Optional[Union[SymbolTableNode, SymbolNode]] = None) -> None:
         if original_ctx:
-            if original_ctx.node and original_ctx.node.get_line() != -1:
-                extra_msg = ' on line {}'.format(original_ctx.node.get_line())
+            if isinstance(original_ctx, SymbolTableNode):
+                node = original_ctx.node
+            elif isinstance(original_ctx, SymbolNode):
+                node = original_ctx
+            else:
+                node = None
+            if node and node.line != -1:
+                extra_msg = ' on line {}'.format(node.line)
             else:
                 extra_msg = ' (possibly by an import)'
```

Comment:
Do we have enough tests for this case? If not you can add one more.

Reply:
I'm not sure when this would happen (when the node's line number is -1). There don't seem to be any tests covering it, so maybe this never happens?

Reply:
Actually, looks like there was one test where this happened. Do you know in what situations this happens? If so, I could add tests for those cases.

Reply:
I think this might happen in case of an error or an unresolved circular import. If you can't find a simple additional test case than ignore this.

Reply:
Hmm, looks like there's already a test case for that situation (`testDuplicateDefFromImport` in `semanal-errors.test`). So I guess we can leave this as-is.